### PR TITLE
Hooked up dialog, puzzle, and return to hub system, cleaned up artwork

### PR DESCRIPTION
**Pull request by JenniBee**
This one's the big pull request I was talking about - I connected the puzzle and dialog system to the doug and clock rooms, added the return to hub after the memories are completed, and cleaned up TimeGentleman's hub artwork a bit.

I also connected the hub up to Escoria (as originally I had it hooked straight up to Godot without the Escoria dependencies - so now the clues from the memories can be used and combined as inventory.

In the Amnesia Fortnight build that I'll be putting up on my github account shortly, the key and the ladder are combined to unlock the silly ending.  In this code, none of that is present - so the items aren't combined.  I'll probably make them do something at some point to demonstrate the code for combining inventory items.

It's not perfect by any means. The use of inventory items on scene items leaves much to be desired, as the text on the verb bar doesn't work properly for it at the moment.  The functionality is present though, it's just that it could use a lot more polish.

### CODE REVIEW

**Comment by fleskesvor** [comment]
Looks good to me. :+1: 

I'm not sure what the changes under globals/ are for, and I think we might be missing something regarding tooltips, since I think they're supposed to be defined in a csv file to facilitate translations. We can look into that later though, once things start to take more shape.

**Comment by JenniBee** [comment]
At the moment, the inventory etc scripts are stored in globals - as per the test scene (ie: inv_can, inv_board, etc).  It is easy to miss them though, so we'll need to move them out eventually.

File: globals/game.gd [hunk 1]
```diff
@@ -28,18 +28,17 @@ func set_mode(p_mode):
...
-		text = text.replace("%2", tr(tt))
-		text = text.replace("%1", tr(current_tool.get_tooltip()))
-	elif obj.inventory:
+
+	#TODO: Fix inventory combination
+	#if current_action == "pick_up" && text != null:
+		#text = tr(current_action + ".combine_id")
+		#text = text.replace("%2", tr(tt))
+		#text = text.replace("%1", tr(current_tool.get_tooltip()))
+	if obj.inventory:
```
Comment (fleskesvor) [inline globals/game.gd:37]: Looks like this file was missing the changes from 9139c386bff4c89136fdda0b2bbc4524ef36bc31 and it breaks interaction in some places.

Comment (fleskesvor) [reply globals/game.gd:37]: Or maybe it doesn't; I'm not sure. I think I made some changes there to support two-click interaction on inventory items. Inventory combine works with this change, but using an inventory item on a non-inventory item seems not to. This also removes localization support, but we haven't done anything to support that yet anyway. Let's just keep this in mind and give it a proper look when we need to.

**Review by fleskesvor — COMMENTED** [review]
Needs testing:
* Inventory on inventory combine (should work)
* Inventory on non-inventory combine (probably doesn't work)
* Localization of combine texts (needs some work)

**Pull request merged**